### ISSUE TITLE
EquatableAnalyzer (CA1067) - false positive, when Equals is sealed

### ISSUE THREAD

**Issue by kindermannhubert** [dotnet/roslyn-analyzers#5497]
### Analyzer

**Diagnostic ID**: [CA1067](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1067): `Override Equals when implementing IEquatable`

### Analyzer source

**SDK**: dotnet\sdk\6.0.100-rc.1.21458.32\Sdks\Microsoft.NET.Sdk

**Version**: 6.0.100-rc.1.21458.32

### Describe the bug

```C#
class Base
{
    public sealed override bool Equals(object? obj) => true;
}

class Derived : Base, IEquatable<Derived> //false positive CA1067 - Equals is sealed and cannot be overriden
{
    public bool Equals(Derived? other) => true;
}
```

### Expected behavior
Do not report when Equals(object?) is sealed.

**Comment by NewellClark** [comment]
Can you describe what lead you to implement `IEquatable` on a derived class and not on its base class? It seems to me that if you are implementing `IEquatable<T>` on a derived class whos base class has sealed `object.Equals(object?)`, then you could easily end up with multiple `Equals` overloads that have different behavior, which would be surprising to developers.

**Comment by kindermannhubert** [comment]
@NewellClark you may want to have something like this:
```C#
abstract class Value : IEquatable<Value>
{
    public override sealed bool Equals(object? obj) => Equals(obj as Value);
    public abstract bool Equals(Value? other);
    //...
}

abstract class Value<T> : Value, IEquatable<Value<T>>
    where T : struct, IEquatable<T>
{
    public T Val { get; }
    public override bool Equals(Value? other) => Equals(other as Value<T>);
    public bool Equals(Value<T>? other) => other != null && Val.Equals(other.Val);
    //...
}
```
